src/boosting/goss.hpp

```diff
@@ -84,20 +85,23 @@ class GOSSStrategy : public SampleStrategy {
 
     CHECK_LE(config_->top_rate + config_->other_rate, 1.0f);
     CHECK(config_->top_rate > 0.0f && config_->other_rate > 0.0f);
-    if (config_->bagging_freq > 0 && config_->bagging_fraction != 1.0f) {
-      Log::Fatal("Cannot use bagging in GOSS");
-    }
+
     Log::Info("Using GOSS");
     balanced_bagging_ = false;
     bag_data_indices_.resize(num_data_);
+    #ifdef USE_CUDA_EXP
```

Comment:
I'm surprised to see new commits referencing `cuda_exp`. `cuda_exp` was completely removed in #5677.

Please update this branch to latest `master` and replace any `cuda_exp` references with `cuda`.